### DISCUSSION TITLE
[Feature] Interpolate Geolocation

### DISCUSSION THREAD

**Discussion by aero31aero** [immich-app/immich#2729] [Feature Request]
### The feature

Often I'd have photos from the same event, but one taken with a DSLR and others with a phone. I was on my way to write a script to do this already, but then found Immich and moved all my files here.... here's the rough idea:

1. Select all photos of [device without GPS radio] -> set A
2. Select all photos of [device with GPS radio] -> set B
3. Select a date range to trim both the above sets to the time you know those two devices were together.
4. For each photo in set A, apply a geolocation by interpolating from the closest 10(?) photos from set B, if they're within a certain time cutoff.

This way, the photos would show up approximately correctly on the map and be easier to search via some later geofencing filters or something.

In my implementation, I was storing another flag in the metadata `gps_interpolated: boolean`, just in case you later wanna filter 'true' data vs 'made-up' data.

This could also help with photos taken from the same device but the first one is taken fast enough that the device didn't yet have a GPS fix.

An example script: [gpscorrelate](https://dfandrich.github.io/gpscorrelate/gpscorrelate.html). This one acts on a set of untagged images and a GPX file. To use this algo, it should be trivial to first create a GPX track from set B to then act on set A.

I'd imagine this would be perfect for the CLI in any case. Here was my imagined command from my notes for when I was writing my tool:

`$ scarfin interpolate-gps --source pixel-2xl --target canon-600d --from 2023-05-23 --to 2023-05-30` 

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by etnoy** [comment]
This is the type of feature I think we won't be adding to immich core and is best implemented using the immich api. You should be able to write an api client that reads and edits the metadata using the endpoints documented in the immich docs
